pkg/volume/util/operationexecutor/operation_executor.go

```diff
@@ -884,11 +912,12 @@ func (oe *operationExecutor) generateUnmountDeviceFunc(
 		unmountDeviceErr := volumeDetacher.UnmountDevice(deviceMountPath)
 		if unmountDeviceErr != nil {
 			// On failure, return error. Caller will log and retry.
-			return fmt.Errorf(
+			err := fmt.Errorf(
 				"UnmountDevice failed for volume %q (spec.Name: %q) with: %v",
 				deviceToDetach.VolumeName,
 				deviceToDetach.VolumeSpec.Name(),
 				unmountDeviceErr)
```

Comment:
@saad-ali  - well, actually I didn't intend to fire events for the volume detach and unmount device and volume unmount for this round, I really wanted to just focus on the failed mounts only for now.  Do you think we need that for this PR or can we follow on with another PR as we see the need?  Also for those events, I'm not clear what runtime object I would pass in for the event, the only time I have a pod object is in the VolumeToMount.